error-handling/src/main/java/com/palantir/remoting2/errors/AbstractServiceException.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting2.errors;
+
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.ws.rs.ext.ExceptionMapper;
+import org.slf4j.Logger;
+
+/**
+ * A base class for defining customized exceptions.
+ */
+public abstract class AbstractServiceException extends RuntimeException {
+
+    private final String errorId = UUID.randomUUID().toString();
+
+    protected AbstractServiceException(String message, @Nullable Throwable cause) {
+        super(message, cause);
+    }
+
+    /** Logs this exception. This method is called by the corresponding {@link ExceptionMapper}. */
+    public abstract void logTo(Logger log);
```

Comment:
Per the discussion on the thread, I don't think you should be taking a logger. Instead, let's just make the constructor of this:

    AbstractServiceException(String message, Param... params)